pandas/core/dtypes/dtypes.py

```diff
@@ -8,6 +8,64 @@
 from .base import ExtensionDtype, _DtypeOpsMixin
 
 
+class Registry(object):
```

Comment:
Without looking at the uses yet, could we simplify this a by just allowing string lookup? Ideally, `registry` would be a simple class holding a dict mapping `dtype.name -> ExtensionDtype`.

Reply:
I think the current code also supports finding the dtype for eg `'interval[int64]'` and not just `interval` (so parametrized versions of the strings)

Reply:
Oh, and I suppose we want that to support `.astype('interval[int64]')`. That's fair...